src/VisualStudio/Core/Def/Implementation/ProjectSystem/AbstractProject.cs

```diff
@@ -624,6 +624,13 @@ private void AddMetadataReferenceCore(VisualStudioMetadataReference reference)
         {
             lock (_gate)
             {
+                if (_metadataReferences.Contains(r => StringComparer.OrdinalIgnoreCase.Equals(r.FilePath, reference.FilePath)))
```

Comment:
Hmm, I forget our conversation again: the classic project systems definitely try to consolidate references so I think this is safe. Did we double check what CPS does? I know they had some consolidation logic but I forget where that landed...?